src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -126,8 +126,9 @@ object MatrixTable {
       case _: TableSpec => fatal(s"file is a Table, not a MatrixTable: '$path'")
     }
     val typ = spec.matrix_type
+    val nCols = spec.colsComponent.read(hc, path, TStruct()).count().toInt
```

Comment:
@cseed is there a way to avoid this?

Reply:
Yeah, you can get the partition counts (there will only be one partition) from the cols table metadata.